File: head-first-design-patterns/Chapter01/C#/Exercises/People/Character.cs
namespace Exercises.People
{
    public abstract class Character
    {
        public IWeaponBehavior weapon;

        public void Fight()
        {
            weapon.UseWeapon();
        }

        public void SetWeapon(IWeaponBehavior wb)
        {
            weapon = wb;
        }
    }
}

File: head-first-design-patterns/Chapter01/Java/Exercises/src/people/Character.java
package people;

public abstract class Character {
    WeaponBehavior weapon;

    public Character() {
        super();
    }

    public void fight() {
        weapon.useWeapon();
    }

    public void setWeapon(WeaponBehavior wb) {
        weapon = wb;
    }
}